### ISSUE TITLE
Error caught on passed in variable to SQS

### ISSUE THREAD

**Issue by oksimon95** [aquasecurity/tfsec#1530]
I am getting a tfsec error caught for `aws-sqs-enable-queue-encryption` for a module, despite the fact that there is a `kms_master_key_id` being passed in to the module. Within the module source, the variable has a default value of `null`, but the implementation of the module has that variable defined and so this error should not be caught.

Any thoughts on what might be causing this? I am currently using v1.1.3. I would be using 1.1.4, which may fix this issue, but homebrew does not yet have 1.1.4

**Comment by oksimon95** [comment]
Sure, here is an example:

The resource to be created, located in the module source:
```
resource "aws_sqs_queue" "default" {
   name              = "sqs_queue"
   kms_master_key_id = var.kms_master_key_id
}
```
variable block of `kms_master_key_id`:
```
variable "kms_master_key_id" {
   default = null
}
```

module implementation that causes `aws-sqs-enable-queue-encryption`:
```
module "queue" {
  source            = "./sqs-module"
  kms_master_key_id = "alias/aws/sqs"
}
```

**Comment by tr-mdaize** [comment]
I do believe that this would still present an error, as we have seen in some of our modules too. The `alias/aws/sqs` key is an internal shared key that is used to encrypt the queue, but tfsec does not have differentiating checks for encrypted vs. encrypted using customer key.

I think the likely recommended path forward would be to separate the `aws-sqs-enable-queue-encryption` into 2 separate checks -- one of high importance checking that the queue is encrypted, and one of slightly lower importance that the user has encrypted the queue with a customer managed KMS key. The shared internal key that AWS provides should be sufficient to satisfy the condition of considering the queue "encrypted".

This logic is likely also relevant for other services where we can encrypt the data using AWS' shared encryption key (which is hardware backed, gets rotated occasionally, etc.) -- eg. RDS, etc.

**Comment by owenrumney** [comment]
> I do believe that this would still present an error, as we have seen in some of our modules too. The `alias/aws/sqs` key is an internal shared key that is used to encrypt the queue, but tfsec does not have differentiating checks for encrypted vs. encrypted using customer key.

This is true, we have recently broken out the check for CMK usage and the check for encryption into two checks for S3 and need to do it for the remainder of the encryption checks

**Comment by oksimon95** [comment]
So in the mean time, I suppose we'll just have to add ignore comments to offending blocks, despite the fact that they will be inaccurate representations of the actual warnings?
